### ISSUE TITLE
FirebaseIndexUpdater.getKeywords splits at accented latin characters?

### ISSUE THREAD

**Issue by increpare** [apsun/uSticker#13]
I have "mühelos" in my file name, and i can only find it by searching for "helos".  Guessing this is to blame:

```
    private fun getKeywords(fileName: String): List<String> {
        // First, split on any non-alphanumeric characters
        val nonSplitAlphaNum = fileName
            .split(Regex("[\\W_]"))
            .filter(String::isNotBlank)

        // Then, split patterns like abc123def to [abc, 123, def]
        return nonSplitAlphaNum.flatMap {
            Regex("[0-9]+|[a-zA-Z]+")
                .findAll(it)
                .map(MatchResult::value)
                .toList()
        }
    }
```

**Comment by increpare** [comment]
(I don't know if Kotlin supports it, but https://stackoverflow.com/questions/3617797/regex-to-match-only-letters/19982853 suggests `\p{L}` to match alphabetic characters in a unicode-friendly way)

**Comment by apsun** [comment]
Hah, such is the life of programmers who assume the universe is in English ;-p. I'll give that pattern a shot, thanks!

**Comment by apsun** [comment]
Fixed in ac79477d637b0ff45b47480be2ec6ada9ffb29fc.
